### PR TITLE
FAQs:[CDS-96084]: FAQs for 29th April

### PR DESCRIPTION
**Pull request by PrakharMartand**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 1]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
 Then when displaying the `failure rate` for the last 30 days you would see `15%` and you would see that your `failure rate` increased by `300%`!
 So the statistic will have a red arrow pointing up marked as a `300%` increment in failure rate. 
 
-This value accurately measures the change in the failure rate, from 5% to 15%. 
\ No newline at end of file
+This value accurately measures the change in the failure rate, from 5% to 15%. 
+
+#### Can one effectively enforce the reconciliation of changes in the template with the consuming pipelines in a forceful manner?
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6946]:
```suggestion
#### Can one effectively enforce the reconciliation of changes in the template with consuming pipelines in a forceful manner?
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 2]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+	stage.template.versionLabel != stableVersion
+
+	# Show a human-friendly error message
+	msg = sprintf(
+		"Stage %s, has template %s, with version %s, it should be version %s",
+		[stage.name, template, stage.template.versionLabel, stableVersion],
+	)
+}
+```
+One can always update the template versions manually, please refer to Harness documentation for more info on: [Reconcile Pipeline Templates](https://developer.harness.io/docs/platform/templates/reconcile-pipeline-templates/)
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6973]:
```suggestion
You can always update the template versions manually. Go to [Reconcile Pipeline Templates](https://developer.harness.io/docs/platform/templates/reconcile-pipeline-templates/) for more information. 
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 3]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+	# Show a human-friendly error message
+	msg = sprintf(
+		"Stage %s, has template %s, with version %s, it should be version %s",
+		[stage.name, template, stage.template.versionLabel, stableVersion],
+	)
+}
+```
+One can always update the template versions manually, please refer to Harness documentation for more info on: [Reconcile Pipeline Templates](https://developer.harness.io/docs/platform/templates/reconcile-pipeline-templates/) 
+
+#### How can one effectively manage remote environments and infrastructures within Harness-SMP?
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6975]:
```suggestion
#### How can one effectively manage remote environments and infrastructures within Harness SMP?
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 4]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+
+To enable remote environments and infrastructures in your Service Management Platform (SMP), follow these implementation guidelines:
+
+- Make sure your SMP is updated to the latest version. 
+- Activate the feature flag (FF) called `CDS_ENV_GITX` on the designated environment.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6980]:
```suggestion
- Activate the feature flag, `CDS_ENV_GITX` on the designated environment.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 5]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+Please refer to the Harness documentation for more info on: [Manage Harness environments and infrastructures from Git](https://developer.harness.io/docs/platform/git-experience/manage-environments-infra-definitions)
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6981]:
```suggestion
Go to [Manage Harness environments and infrastructures from Git](https://developer.harness.io/docs/platform/git-experience/manage-environments-infra-definitions) for more information.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 6]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+
+#### Does Harness support the use of OpenID Connect(OIDC) for connecting to various systems such as Amazon Web Services(AWS) and Google Cloud Platform (GCP)?
+
+Yes, our current capability includes support for Google Cloud Platform (GCP) and Amazon Web Services (AWS) in terms of OIDC integration.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6985]:
```suggestion
Yes, we currently support OIDC integration for Google Cloud Platform (GCP) and Amazon Web Services (AWS).
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 7]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+Within the next three months, we aim to further enhance this support, facilitating authentication, short-lived token acquisition based on Harness context, and various operational tasks like deployment, builds, or secret retrieval within the respective cloud provider environments.
+GCP integration and AWS integration is currently available for utilization.
+Please refer to Harness documentation for more info on: [GCP Connectors with OIDC](https://developer.harness.io/docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference/#use-openid-connect-oidc) and [AWS Connector with OIDC](https://developer.harness.io/docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference/#credentials)
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6988]:
```suggestion

```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 8]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+
+#### Does Harness enforce any policy that denies the fetching of the latest tag of an image, as indicated by the error message : `admission webhook: <webhook-name> denied the request. Validation error: An image tag is required. rule require-image-tag failed at path`?
+
+Harness does not enforce any policy regarding fetching the latest tag of images.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6992]:
```suggestion
Harness does not enforce any policy regarding fetching the latest image tags.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 9]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+To debug this error, consider the following implementations:
+- Ensure the correct image tag is specified in the YAML configuration.
+- Refer to the Harness documentation for the correct expression usage, particularly regarding the retrieval of the latest published tag: [Latest Published Tag](https://developer.harness.io/kb/continuous-delivery/articles/last-published-tag/)
+
+#### Does Harness support coverage testing in Python? Additionally, can coverage testing be used alongside split testing?
+
+Yes, Harness supports code coverage testing alongside split testing in Python.
+Please refer to Harness documentation on: [Code coverage in Harness CI](https://developer.harness.io/docs/continuous-integration/use-ci/run-tests/code-coverage) and [Split tests (parallelism) in Harness CI](https://developer.harness.io/docs/continuous-integration/use-ci/run-tests/speed-up-ci-test-pipelines-using-parallelism)
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7000]:
```suggestion
Go to [Code coverage in Harness CI](https://developer.harness.io/docs/continuous-integration/use-ci/run-tests/code-coverage) and [Split tests (parallelism) in Harness CI](https://developer.harness.io/docs/continuous-integration/use-ci/run-tests/speed-up-ci-test-pipelines-using-parallelism) for more information.
```

Comment (roshnisarangadharan) [reply kb/continuous-delivery/continuous-delivery-faqs.md:7000]: This is a candidate for CI FAQ.

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 10]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+Please refer to Harness documentation for more info on: [Git Repository Merge PRs](https://developer.harness.io/docs/code-repository/pull-requests/merge-pr/)
+
+#### Does Harness support Test Intelligence (TI) for Python in CI?
+
+Yes, this feature is currently behind the FF: `CI_PYTHON_TI`.
+To enable TI for Python use three steps:
+- Add the Run Tests step. 
+- Trigger test selection.
+- (Optional) Add test splitting.
+Please refer to Harness documentation for more info on: [Enable TI for Python](https://developer.harness.io/docs/continuous-integration/use-ci/run-tests/test-intelligence/ti-for-python/)
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7015]: Move these to code repo and CI FAQs

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 11]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+
+#### Does Harness support storing the Terraform Plan on the Harness Delegate temporarily?
+
+Yes, Users can now store the terraform plan on the delegate and leverage it in the apply step. This now by passes the restriction to store the plan in a secrets manager and lets users store it locally.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7019]:
```suggestion
Yes, Users can now store the terraform plan on the delegate and leverage it in the apply step. This now bypasses the restriction to store the plan in a secrets manager and let users store it locally.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 12]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+This feature is behind the FF: `CDS_STORE_TERRAFORM_PLAN_FILE_LOCALLY_ON_DELEGATE` and on delegate version `827xx or more`.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7020]:
```suggestion
This feature is behind the feature flag, `CDS_STORE_TERRAFORM_PLAN_FILE_LOCALLY_ON_DELEGATE`. Harness Delegate version 827xx or later is required for this feature.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 13]
```diff
@@ -6941,4 +6941,81 @@ Days 31 - 60 `failure rate`: 15%
...
+Please refer to Harness documentation for more info: [Store Terraform Plan on Harness Delegate](https://developer.harness.io/docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-plan-step/#store-terraform-plan-on-harness-delegate) and [Demo Video](https://www.loom.com/share/bc5a4f382d584b228b4ea2c82eb94a7c?sid=b9fac5c3-c11b-4f50-acff-f4fd2b3cc83a)
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7021]:
```suggestion
Go to [Store Terraform Plan on Harness Delegate](https://developer.harness.io/docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-plan-step/#store-terraform-plan-on-harness-delegate) and [Demo Video](https://www.loom.com/share/bc5a4f382d584b228b4ea2c82eb94a7c?sid=b9fac5c3-c11b-4f50-acff-f4fd2b3cc83a) for more information.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md (hunk 10 above)
Comment (PrakharMartand) [reply kb/continuous-delivery/continuous-delivery-faqs.md]: Removed as of now, will have it in next PR

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 14]
```diff
@@ -6941,4 +6941,70 @@ Days 31 - 60 `failure rate`: 15%
...
+
+#### Does the `/rebase` command functional when used from the PR comment?
+
+Yes, Harness supports the `/rebase` command from PR comments.
+If encountering a failure, please verify that there are no conflicts present in the git branch being utilized.
+Please refer to Harness documentation for more info on: [Git Repository Merge PRs](https://developer.harness.io/docs/code-repository/pull-requests/merge-pr/)
+
+#### Does Harness support storing the Terraform Plan on the Harness Delegate temporarily?
+
+Yes, Users can now store the terraform plan on the delegate and leverage it in the apply step. This now bypasses the restriction to store the plan in a secrets manager and let users store it locally.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7008]:
```suggestion
Yes, users can now store the terraform plan on the delegate and leverage it in the apply step. This now bypasses the restriction to store the plan in a secrets manager and let users store it locally.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 15]
```diff
@@ -6941,4 +6941,70 @@ Days 31 - 60 `failure rate`: 15%
...
+
+#### Does Harness support coverage testing in Python? Additionally, can coverage testing be used alongside split testing?
+
+Yes, Harness supports code coverage testing alongside split testing in Python.
+Go to [Code coverage in Harness CI](https://developer.harness.io/docs/continuous-integration/use-ci/run-tests/code-coverage) and [Split tests (parallelism) in Harness CI](https://developer.harness.io/docs/continuous-integration/use-ci/run-tests/speed-up-ci-test-pipelines-using-parallelism) for more information.
+
+#### Does the `/rebase` command functional when used from the PR comment?
+
+Yes, Harness supports the `/rebase` command from PR comments.
+If encountering a failure, please verify that there are no conflicts present in the git branch being utilized.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7003]:
```suggestion
If encountering a failure, verify that there are no conflicts present in the git branch being utilized.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 16]
```diff
@@ -6941,4 +6941,70 @@ Days 31 - 60 `failure rate`: 15%
...
+Please refer to Harness documentation for more info on: [Git Repository Merge PRs](https://developer.harness.io/docs/code-repository/pull-requests/merge-pr/)
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7004]:
```suggestion
Go to [Git Repository Merge PRs](https://developer.harness.io/docs/code-repository/pull-requests/merge-pr/) for more information.
```

**Review by roshnisarangadharan — APPROVED** [review]
Let me know when to merge

**Review by krishi0408 — APPROVED** [review]

**Pull request merged**